pkg/controller/nodeipam/legacyprovider.go

```diff
@@ -62,10 +63,10 @@ func createLegacyIPAM(
 	}
 	ipamc, err := ipam.NewController(cfg, kubeClient, cloud, cidr, serviceCIDR, nodeCIDRMaskSizes[0])
 	if err != nil {
-		klog.Fatalf("Error creating ipam controller: %v", err)
+		return nil, errors.Wrapf(err, "Error creating ipam controller")
 	}
 	if err := ipamc.Start(nodeInformer); err != nil {
-		klog.Fatalf("Error trying to Init(): %v", err)
+		return nil, errors.Wrapf(err, "Error trying to Init()")
```

Comment:
updated here as well.

- other `klog.Fatal()` are in some other files under `pkg/controller/nodeipam/` 
<img width="449" alt="image" src="https://user-images.githubusercontent.com/2010320/222855676-d307f5d7-f9b1-4286-a1f4-6b91f6641c98.png">